### PR TITLE
add multiselect field option

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
The api needs to accept the multiselect boolean value in the Field Struct

#### Ticket Link
n/a

### CODE REVIEW

File: server/api/field.go
Comment (levb) [inline server/api/field.go:39]: @jfrerich 1/5 `SelectMultiselect` is consistent, but redundant a name. Perhaps, just `Multiselect`, or even `Multi`?

@mickmister this (multiple selection support) seems like a bigger deal than just adding the flag for it, or are other pieces in place? Autocomplete worries me in particular.

**Review by levb — DISMISSED** [review]

File: server/api/field.go
Comment (jfrerich) [reply server/api/field.go:39]: I think the `Select` prefix is valuable because it is grouping Field values that are explicitly related to a select form.  I suppose a `SelectIsMulti` my be more intention revealing, implying a boolean option for a select field

Comment (jfrerich) [reply server/api/field.go:39]: For reference, I have not tested this for autocompletion, but have multiselect fields working in the zendesk app in this PR https://github.com/mattermost/mattermost-app-zendesk/pull/21

**Comment by mickmister** [comment]
@jfrerich I'm thinking we hold off on merging this until it is supported in the client. What do you think?

File: server/api/field.go
Comment (mickmister) [reply server/api/field.go:39]: > but have multiselect fields working in the zendesk app in this PR

Are you saying that it is already supported in the webapp?

Comment (jfrerich) [reply server/api/field.go:39]: Yes. There is an oddity where the multiselect opens with an empty value already in the select box, but it is working.

Comment (jfrerich) [reply server/api/field.go:39]: image

Comment (jfrerich) [reply server/api/field.go:39]: Here is the multiselect with multiple selected fields

image

Comment (mickmister) [reply server/api/field.go:39]: > Yes. There is an oddity where the multiselect opens with an empty value already in the select box, but it is working.

Can you make a ticket for this if it is not resolved?

Comment (mickmister) [reply server/api/field.go:39]: 1/5 `SelectIsMulti` sounds good to me

**Review by mickmister — APPROVED** [review]

File: apps/field.go
Comment (larkox) [inline apps/field.go:64]: Since everything is now heavily documented, should we add here some documentation too?

**Review by larkox — APPROVED** [review]

**Comment by hanzei** [comment]
@jfrerich Is this PR needed for V1 of either of the two apps?

**Review by DHaussermann — APPROVED** [review]
Tested and passed.
- Verified with Zendesk App 
- Control is working as expected
- Data sent is accurate 
- Follow-up ticket needed for RN client support

LGTM!

**Comment by jfrerich** [comment]
@levb, this is ready to merge. Dylan and I tested again today! For some reason, I do not have merge permissions.

**Comment by hanzei** [comment]
@jfrerich Heads up that docs need to be submitted to https://developers.mattermost.com/integrate/apps/

**Comment by hanzei** [comment]
@levb Are you fine with merging this PR?=

**Comment by jfrerich** [comment]
@hanzei we discussed in the meeting this morning that we will merge this PR, but will not yet advertise it in the docs for the following reasons.
* command form is not yet handled
* mobile has to be handled by the app

For zendesk, which does use this function, I don't have multiselect via a slash command and will handle the mobile case in the app.
cc: @mickmister

**Comment by jfrerich** [comment]
Here is the PR for React Native implementation.  I've added a note to add the developer docs when that Ticket is worked on.

https://mattermost.atlassian.net/browse/MM-34194

**Comment by hanzei** [comment]
@jfrerich Sounds good to me :+1:

**Pull request merged**
